skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -529,46 +509,45 @@ public function executeEditBox() {
 
 		// Editing: foo
 		$this->displaytitle = true;
-		$this->title = wfMsg( $msg, htmlspecialchars( $wgTitle->getPrefixedText() ) );
+		$this->title = wfMessage( $msg, $this->wg->Title->getPrefixedText() )->parse();
 
 		// back to article link
-		$this->subtitle = Wikia::link( $wgTitle, wfMsg( 'oasis-page-header-back-to-article' ), [ 'accesskey' => 'c' ], [ ], 'known' );
+		$this->subtitle = Linker::linkKnown( $this->wg->Title, wfMessage( 'oasis-page-header-back-to-article' )->escaped(), [ 'accesskey' => 'c' ] );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)